.github/workflows/build.yaml

```diff
@@ -1,14 +1,16 @@
 name: Build
 on:
-  pull_request:
-    types:
-      - opened
-      - synchronize
+  workflow_call:
   push:
-    branches:
-      - master
     tags:
-      - '*-nightly-*'
+    - 'nightly-*'
+    branches:
+    - master
+  pull_request:
```

Comment:
I added `reopened` to harmonise the trigger with the `style.yaml`, `unit-tests.yaml` workflows.